extensions/emmet/src/abbreviationActions.ts

```diff
@@ -436,14 +443,20 @@ function expandAbbr(input: ExpandAbbreviationInput): string | undefined {
 		let expandedText = helper.expandAbbreviation(input.abbreviation, expandOptions);
 
 		if (input.textToWrap) {
+			// Fetch innermost element in the expanded abbreviation
+			let wrappingEnd = expandedText.substring(expandedText.indexOf('$TM_SELECTED_TEXT'));
+			let tagName = wrappingEnd.substring(wrappingEnd.indexOf('/') + 1, wrappingEnd.indexOf('>'));
+
+			// If wrapping with a block element, insert newline and expand again
+			if (inlineElements.indexOf(tagName) === -1 && input.textToWrap.length === 1) {
+				if (expandOptions['text'][0].indexOf('\n') === -1) {
+					expandOptions['text'][0] = '\n\t' + expandOptions['text'][0] + '\n';
+				}
+				expandedText = helper.expandAbbreviation(input.abbreviation, expandOptions);
```

Comment:
Why expand again? Why not just add the `\n\t` and `t` around `$TM_SELECTED_TEXT` in the expanded text?